zerver/tests/test_event_queue.py

```diff
@@ -128,456 +119,421 @@ def test_stream_watchers(self) -> None:
-                get_events,
-                user,
-                {
-                    "apply_markdown": orjson.dumps(True).decode(),
-                    "client_gravatar": orjson.dumps(True).decode(),
-                    "event_types": orjson.dumps(["message"]).decode(),
-                    "user_client": "website",
-                    "dont_block": orjson.dumps(True).decode(),
-                },
-            )
-            self.assert_json_success(result)
-            queue_id = orjson.loads(result.content)["queue_id"]
-            return access_client_descriptor(user.id, queue_id)
-
-        def destroy_event_queue(user: UserProfile, queue_id: str) -> None:
-            result = self.tornado_call(cleanup_event_queue, user, {"queue_id": queue_id})
-            self.assert_json_success(result)
-
-        def assert_maybe_enqueue_notifications_call_args(
-            args_dict: Collection[Any],
-            message_id: int,
-            user_id: int,
-            **kwargs: Any,
-        ) -> None:
-            expected_args_dict = self.get_maybe_enqueue_notifications_parameters(
-                user_id=user_id,
-                acting_user_id=iago.id,
-                message_id=message_id,
-                **kwargs,
+    def assert_maybe_enqueue_notifications_call_args(
+        self,
+        args_dict: Collection[Any],
+        message_id: int,
+        user_id: int,
+        **kwargs: Any,
+    ) -> None:
+        expected_args_dict = self.get_maybe_enqueue_notifications_parameters(
+            user_id=user_id,
+            acting_user_id=self.example_user("iago").id,
+            message_id=message_id,
+            **kwargs,
+        )
+        self.assertEqual(args_dict, expected_args_dict)
+
+    def change_subscription_properties(
+        self,
+        user_profile: UserProfile,
+        stream: Stream,
+        sub: Subscription,
+        properties: Dict[str, bool],
+    ) -> None:
+        for property_name, value in properties.items():
+            do_change_subscription_property(
+                user_profile, sub, stream, property_name, value, acting_user=None
             )
-            self.assertEqual(args_dict, expected_args_dict)
 
-        client_descriptor = allocate_event_queue(user_profile)
+    def setUp(self) -> None:
+        self.user_profile = self.example_user("hamlet")
+        self.cordelia = self.example_user("cordelia")
+        self.user_profile.enable_online_push_notifications = False
+        self.user_profile.save()
+        self.iago = self.example_user("iago")
+        self.client_descriptor = self.allocate_event_queue(self.user_profile)
+        self.assertTrue(self.client_descriptor.event_queue.empty())
+
+        self.stream = get_stream("Denmark", self.user_profile.realm)
+        self.sub = Subscription.objects.get(
+            user_profile=self.user_profile,
+            recipient__type=Recipient.STREAM,
+            recipient__type_id=self.stream.id,
+        )
```

Comment:
What's the thinking behind fetching self.sub / self/stream in the `setUp` method?

The thing I worry about is these fetches having stale data if the test itself mutates these objects in its setup code, which I know a few of them do (to mute the stream, or whatever).

Reply:
Reading the code a bit, I suspect we could just fetch these inside `change_subscription_properties` rather than here, and avoid that issue.

Reply:
Done